From ea2fe3fd348478544b42c74b28860b29f8fc1bd7 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Wed, 30 Oct 2019 09:21:36 +0000
Subject: [PATCH] systemd: fix avc denials for systemd-ask-password

Fix errors:
systemd-tty-ask-password-agent[1594]: Failed to determine devices of /dev/console: Permission denied
systemctl[2568]: Failed to stop systemd-ask-password-plymouth.path: Access denied
systemctl[2568]: Failed to stop systemd-ask-password-plymouth.service: Access denied

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/init.te    |  2 +-
 policy/modules/system/systemd.te | 11 ++++++++++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/policy/modules/system/init.te b/policy/modules/system/init.te
index fc83705..779e6ca 100644
--- a/policy/modules/system/init.te
+++ b/policy/modules/system/init.te
@@ -971,7 +971,7 @@ ifdef(`enabled_mls',`
 ')
 
 ifdef(`init_systemd',`
-	allow initrc_t init_t:system { start status reboot halt reload };
+	allow initrc_t init_t:system { start status reboot halt reload stop };
 
 	manage_files_pattern(initrc_t, initrc_lock_t, initrc_lock_t)
 	files_lock_filetrans(initrc_t, initrc_lock_t, file)
diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 6dd8f7a..96e11d0 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -937,7 +937,7 @@ optional_policy(`
 # systemd_passwd_agent_t local policy
 #
 
-allow systemd_passwd_agent_t self:capability { chown sys_tty_config dac_override };
+allow systemd_passwd_agent_t self:capability { chown sys_tty_config dac_override sys_resource };
 allow systemd_passwd_agent_t self:process { setfscreate setsockcreate signal };
 allow systemd_passwd_agent_t self:unix_dgram_socket create_socket_perms;
 
@@ -955,6 +955,15 @@ dev_read_generic_files(systemd_passwd_agent_t)
 dev_write_generic_sock_files(systemd_passwd_agent_t)
 dev_write_kmsg(systemd_passwd_agent_t)
 
+mls_file_write_all_levels(systemd_passwd_agent_t)
+mls_file_read_all_levels(systemd_passwd_agent_t)
+
+kernel_read_kernel_sysctls(systemd_passwd_agent_t)
+corecmd_search_bin(systemd_passwd_agent_t)
+
+dev_write_sysfs_dirs(systemd_passwd_agent_t)
+dev_read_sysfs(systemd_passwd_agent_t)
+
 files_read_etc_files(systemd_passwd_agent_t)
 files_map_etc_files(systemd_passwd_agent_t)
 
-- 
2.7.4

